### PR TITLE
Add Sort Order Support to /terms API Request

### PR DESCRIPTION
**Pull request by billmurrin**
## Description
This changes adds *order* as an option when conducting an */search/universal/relative/terms* API query. This will allow the web interface to request ascending ordered term aggregations via the REST API. 

## Motivation and Context
* PR #3540 added sort order support to the Searches.terms method for sort order. This PR adds it to the Graylog API so that the web interface can take advantage of it via REST queries. 
* Right now, a term aggregation returned from a REST query can only be ordered Ascending in a generic manner since the underlying query is ordered in a descending manner.
* This PR fixes a bug I am experiencing in my QuickValuesPlus Widget Plugin [More Info](https://github.com/billmurrin/graylog-plugin-quickvaluesplus-widget/issues/7)

## How Has This Been Tested?
* I compiled the Graylog 2.3.0 SNAPSHOT with the incorporated changes and then replaced the graylog.jar file under /opt/graylog/server. Once the server loaded, I validated the latest SNAPSHOT version "2.3.0-SNAPSHOT+252f9fe" was loaded. I went into the API Browser on the node and validated under "GET /search/universal/relative/terms" that the *order* field was present. 
* I tested it with values for *asc*, *desc*, and blank and was able to see that the underlying JSON query contained the proper sort syntax. The results also indicated the query was providing the correct data.
* No unit testing on the code change was conducted.
* No errors were observed.

## Screenshots (if appropriate):
alt text

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/search/RelativeSearchResource.java [hunk 1]
```diff
@@ -188,11 +190,20 @@ public TermsResult termsRelative(
...
             @ApiParam(name = "size", value = "Maximum number of terms to return", required = false) @QueryParam("size") int size,
             @ApiParam(name = "range", value = "Relative timeframe to search in. See search method description.", required = true) @QueryParam("range") int range,
-            @ApiParam(name = "filter", value = "Filter", required = false) @QueryParam("filter") String filter) {
+            @ApiParam(name = "filter", value = "Filter", required = false) @QueryParam("filter") String filter,
+            @ApiParam(name = "order", value = "Sorting (field:asc / field:desc)", required = false) @QueryParam("order") String order) {
         checkSearchPermission(filter, RestPermissions.SEARCHES_RELATIVE);
 
+        final Sorting.Direction sortOrder;
+
+        if (isNullOrEmpty(order)) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/search/RelativeSearchResource.java]: You could use [`buildSorting()`](https://github.com/Graylog2/graylog2-server/blob/2.2.3/graylog2-server/src/main/java/org/graylog2/rest/resources/search/SearchResource.java#L226-L237) here, also see [`Sorting`](https://github.com/Graylog2/graylog2-server/blob/2.2.3/graylog2-server/src/main/java/org/graylog2/indexer/searches/Sorting.java).

**Comment by billmurrin** [comment]
@joschi. Added buildSorting method support. In order to return the Direction from the Sorting class (Searches.terms expects a Sorting.Direction object) I added a getDirection method to the Sorting class. Your review is appreciated.

**Comment by joschi** [comment]
Superseded by #3829

**Comment by joschi** [comment]
@billmurrin Thank you very much for your contribution! We've merged the rebased PR into master (6e5d70e15f7920f39718c82c395a29d42c38fe8a).

**Pull request closed without merging**